app/queries/facility_analytics_query.rb

```diff
@@ -37,28 +37,19 @@ def registered_patients_by_period
   end
 
   def follow_up_patients_by_period
-    #
-    # this is similar to what the group_by_period query already gives us,
-    # however, groupdate does not allow us to use these "groups" in a where clause
-    # hence, we've replicated its grouping behaviour in order to remove the patients
-    # that were registered prior to the period bucket
-    #
-    date_truncate_string =
-      "(DATE_TRUNC('#{@period}', blood_pressures.recorded_at::timestamptz AT TIME ZONE '#{Groupdate.time_zone || 'Etc/UTC'}'))"
-
     @follow_up_patients_by_period ||=
-      BloodPressure
-        .left_outer_joins(:user)
-        .left_outer_joins(:patient)
-        .joins(:facility)
-        .where(facility: @facility)
-        .where(deleted_at: nil)
-        .group('users.id')
-        .group_by_period(@period, 'blood_pressures.recorded_at')
-        .where("patients.recorded_at < #{date_truncate_string}")
-        .order('users.id')
-        .distinct
-        .count('patients.id')
+      Patient
```

Comment:
This is an ugly hacky query, but we have intentionally not spent too much effort improving it. It however, produces the correct results.